### ISSUE TITLE
Show all single and double quotes in footnotes

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord-dev/ts-desktop#1111]
Android displays all quote marks in the footnotes and it makes it much easier to understand. I think it's worth the effort to do that on desktop too.

**Comment by cdwhitfield73** [comment]
All "actual" quote marks (single or double) that are in the text of the footnote are being displayed on Desktop.  The extra quotes you are seeing in Android are being added by the app based on "char" tags with an "fqa" type.  Desktop is not currently doing any formatting based on these tags.  If desired, we could add code to parse and format based on these tags (there are several different types of tags), but we would need a complete list of what tag types are out there and what you want done with them.  Also, there seems to be some tags in the source that should not be there and may produce undesirable results if used for formatting.  Waiting on Koz to talk to Christine and Klappy before proceeding.

**Comment by ChrisJarka** [comment]
Digging also shows that /fqb notation in footnotes are also not rendering correctly.
For 1CH 1:5-7 the footnote in the USFM file is:

\f + \ft Some versions have \fqa Diphath \fqb instead of \fqa Riphath \fqb . However, \fqa Diphath \fqb was probably a mispelling. Many ancient copies correct it to \fqa Riphath \fqb in order to make it agree with the same name in Gen. 10:2. \f*

But in tS-D v11 build 110 it has "Some version have Diphath Riphath Diphath Riphath" as the footnote.

**Comment by cckozie** [comment]
There are several issues related to this problem. There are "fqb" tags in that source and the related text does not show up in the source data on the device. See second footnote in 1ch.
Christine is going to get with the source team to map a direction. She and I agree the best would probably be to not use the tags, but rather always use the actual quote characters.
We'll do nothing on the tS side until the issue on the source side gets figured out.

**Comment by cdwhitfield73** [comment]
This is a data issue that occurs before we (the app or the Door43 client) even get it.  I have made contact with the Web Team about this issue and they are looking into a fix for it.  It was suggested by them that the fix may be having the content team change the fqb tags into fqa* tags, which is what their code is apparently expecting to signify the end of a quote.

**Comment by cdwhitfield73** [comment]
https://github.com/unfoldingWord-dev/door43.org/issues/483 has been created for the Web Team to evaluate the problem.

**Comment by cdwhitfield73** [comment]
The web team issue referenced above (483) is complete.  The "fqb" tags have been replaced with "fqa*" tags, which allows the data to be parsed correctly without losing text (like the example above, in 1Ch 1:5-7).  All text in footnotes will now show up in Desktop, assuming that we update the data package.  However, there is still nothing in Desktop that is parsing "fqa" tags and turning them into quote marks around the tagged text.  Do we want to do this?  Or leave it as is?  Christine's suggestion was for the content group to not use the tags and just put in the quote marks, but this has not been done.  If it is done, tS-D will handle it correctly.

So, two separate issues here:
1. update the data that comes packaged with 11.1 to include the fixes for the tags (may be other changes/fixes/additions to the data as well)
2. decide if any format parsing needs to happen to create quote marks out of fqa tags

**Comment by cckozie** [comment]
@cdwhitfield73 I'll ping Christine and see what she want's to do. My bias is to do nothing on the tS side and let them "fix" it in the source. I did a Google search for fqa and found this interesting hit:
https://github.com/Door43/ulb-en/issues/283

**Comment by cdwhitfield73** [comment]
The packaged data has been updated as of 3-29-17.  Nothing has been changed to render quote marks from tags, but the text will now all display (after you download the updates for the resource container)

**Comment by ChrisJarka** [comment]
Sorry, not fixed. Here is tS-Android v11.0 Build 167 with a fresh update of the EN ULB for 1CH installed and being used.
For the footnote of 1CH 1:6 that is:
\v 6 The sons of Gomer were Ashkenaz, Riphath, and Togarmah.
\f + \ft Some versions have \fqa Diphath \fqa* instead of \fqa Riphath \fqa* . However, \fqa Diphath \fqa* was probably a mispelling. Many ancient copies correct it to \fqa Riphath \fqa* in order to make it agree with the same name in Gen. 10:2. \f*

This is what shows in tS-Android

![bad footnote fqa](https://cloud.githubusercontent.com/assets/9850185/24509581/b8902140-1534-11e7-9d30-583566bc6833.png)

**Comment by cdwhitfield73** [comment]
@ChrisJarka This issue is for Desktop only.  The footnotes appear correctly in Build 129 of Desktop.  This will need to be addressed in Android as a separate issue in their repo.

**Comment by ChrisJarka** [comment]
Sorry @cdwhitfield73 I keep forgetting to look where I am!

**Comment by cdwhitfield73** [comment]
Desktop will now render all fqa tags as double quote marks around the tagged text.  Any single or double quotes actually in the text will also render.

**Comment by cckozie** [comment]
@cdwhitfield73 Just need to verify that it is expected to not render the footnotes correctly until downloading new sources. In the latest Android build the footnotes are rendered correctly, but in the latest desktop build there are still some problems before downloading sources.
![image](https://cloud.githubusercontent.com/assets/8171759/24554704/226e8934-15f4-11e7-8115-3d21298b1ef0.png)
![image](https://cloud.githubusercontent.com/assets/8171759/24554717/29d3fa74-15f4-11e7-9305-c526532db675.png)

**Comment by cdwhitfield73** [comment]
When upgrading to Build 129 or 130, you will get all the new sources in the "packaged" folder (where they are in the zipped format).  But any resource containers that you already have opened in your "local" folder will not be automatically overwritten.  This was by design.  If you go into the source modal for any of your previous projects, you will probably see the "update" icon now next to your selected ULB sources.  The footnote changes will not show up until you run the "update",  or download them again from the download sources page.  Another option would be to delete your local resource container folder before starting the app, which would cause it to re-open all necessary containers from the zipped folder and then you would have all the newest data.  That would simulate a user who goes straight from version 10 to version 11.1.  So, yes, it sounds like it is working as expected.  We decided the user should have the option of continuing to use the older version of a source if they had already started a project with it and only update by their choosing.

**Comment by cckozie** [comment]
@cdwhitfield73 Am I correct in assuming that when upgrading from v10 to 11.1 I should get the new text source since it is now in resource containers?

**Comment by cdwhitfield73** [comment]
Yes and no.  For the average user going from only ever having 10 or below and upgrading to 11 or 11.1, it will create the resource container folder and open up the most current containers for the existing projects.  For you, as a tester, there is a possibility it won't.  If you have run 11 or 11.1 on the machine, and then go backwards to 10 and then go back to 11.1, the resource container folder will still be there from before, in which case, it will not overwrite any existing containers that were opened before.  To simulate a real life scenario, whenever you go back to 10, you should manually delete the resource container folder in the library.  In that case, then yes, your assumption is correct.

**Comment by cckozie** [comment]
I started the upgrade testing by deleting the hidden tS folder and did all of the upgrades as they would be done in the field, never going backwords. So I think what I saw was as expected. When upgrading to 11.1 from either 9 or 10 I saw the footnotes rendered properly and did not see the update arrow on the English ULB source. When I upgraded from 11 to 11.1, the footnotes were not rendered properly until I updated the English ULB.
